From 73b36dd9c5c8275fdb3cb2b4bfb365968f520e5c Mon Sep 17 00:00:00 2001
From: Friedemann Kleint <Friedemann.Kleint@qt.io>
Date: Fri, 19 Dec 2025 07:54:49 +0100
Subject: [PATCH] Partially revert "PySide6: Add a conversion tuple->QVariantList"

This reverts commits c85f92334518c61e2c455efb12c659f37c4c1196
and 9086240ff5766b996fd5dd0020698932d38da039.

The feature causes unexpected behavior when storing tuples
in QVariants, for example in QCombox item data. Revert the
entire change and its amendment fix for named tuples.
The test is kept and adapted, though.

Task-number: PYSIDE-3206
Task-number: PYSIDE-3244
Fixes: PYSIDE-3256
Pick-to: 6.10

Change-Id: If486758d6552bf8fa30cf945d141a0f5a44ce805
Reviewed-by: Cristian Maureira-Fredes <cristian.maureira-fredes@qt.io>
---

diff --git a/sources/pyside6/PySide6/QtCore/typesystem_core_common.xml b/sources/pyside6/PySide6/QtCore/typesystem_core_common.xml
index f001178..bf04e04 100644
--- a/sources/pyside6/PySide6/QtCore/typesystem_core_common.xml
+++ b/sources/pyside6/PySide6/QtCore/typesystem_core_common.xml
@@ -336,8 +336,6 @@
             <add-conversion type="SbkObject" file="../glue/qtcore.cpp" snippet="conversion-sbkobject"/>
             <add-conversion type="PyDict" check="PyDict_CheckExact(%in)" file="../glue/qtcore.cpp" snippet="conversion-pydict"/>
             <add-conversion type="PyList" check="PyList_Check(%in)" file="../glue/qtcore.cpp" snippet="conversion-pylist"/>
-            <add-conversion type="PyTuple" check="PyTuple_CheckExact(%in)"
-                            file="../glue/qtcore.cpp" snippet="conversion-pylist"/>
             <add-conversion type="PyObject" file="../glue/qtcore.cpp" snippet="conversion-pyobject"/>
         </target-to-native>
     </conversion-rule>
diff --git a/sources/pyside6/tests/QtCore/qobject_property_test.py b/sources/pyside6/tests/QtCore/qobject_property_test.py
index 9d2bd2c..e0a8044 100644
--- a/sources/pyside6/tests/QtCore/qobject_property_test.py
+++ b/sources/pyside6/tests/QtCore/qobject_property_test.py
@@ -106,9 +106,10 @@
 
 
 class VariantPropertyTest(unittest.TestCase):
-    """Test QVariant conversion in properties and signals (PYSIDE-3206, PYSIDE-3244).
-       It uses a property of list type that is passed a QVariantList
-       with various element types when using QObject.setProperty()."""
+    """Test QVariant conversion in properties and signals (PYSIDE-3256,
+       PYSIDE-3244, PYSIDE-3206 [open]). It uses a property of list type
+       that is passed a QVariantList with various element types when
+       using QObject.setProperty()."""
 
     def testIt(self):
         to = TestVariantPropertyObject()
@@ -123,11 +124,11 @@
         to.setProperty("testProperty", [{"key": 42}])
         self.assertEqual(type(to.get_property()[0]), dict)
 
-        # PYSIDE-3206 (DBus): Convert a tuple to a list
+        # Tuple (PYSIDE-3256)
         to.setProperty("testProperty", [(1, 2)])
-        self.assertEqual(type(to.get_property()[0]), list)
+        self.assertEqual(type(to.get_property()[0]), tuple)
 
-        # PYSIDE-324: The tuple conversion must not occur for named tuples
+        # Named Tuple (PYSIDE-3244)
         to.setProperty("testProperty", [Point(1, 2)])
         self.assertEqual(type(to.get_property()[0]), Point)
 
